tests/test_kaggle_cache_dataset_download.py

```diff
@@ -0,0 +1,86 @@
+import os
+from unittest import mock
+
+import requests
+
+import kagglehub
+from kagglehub.config import DISABLE_KAGGLE_CACHE_ENV_VAR_NAME
+from kagglehub.env import KAGGLE_DATA_PROXY_URL_ENV_VAR_NAME
+from tests.fixtures import BaseTestCase
+
+from .server_stubs import jwt_stub as stub
+from .server_stubs import serv
+
+INVALID_ARCHIVE_DATASET_HANDLE = "invalid/invalid/invalid/invalid/invalid"
+VERSIONED_DATASET_HANDLE = "sarahjeffreson/featured-spotify-artiststracks-with-metadata/versions/1"
```

Comment:
Orthogonal to this PR but the fact that the handle versioning is different from a model may be bad UX.

For models, the version is added with a `/{VERSION_NUMBER}` suffix, not `/versions/{VERSION_NUMBER}`.

However, I do understand that we chose it to ensure it aligns with the URL format.

Curious to have @goeffthomas / @jplotts thoughts on this.

Reply:
Yeah, this is a bit rough since Models routing is now very different from Datasets. Would it be a bad idea to support both for Datasets? I know for Models, the UI now has the shortened URLs to match the handles. Being able to support both would mean that if we did want Datasets to follow that pattern, we could do it at our leisure with no impact to `kagglehub`.